client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -195,19 +201,18 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( cart.create_new_blog && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
+		} else if ( cart.create_new_blog && ! cartItems.hasPlan( cart ) ) {
+			if ( isEmpty( receipt.failed_purchases ) ) {
```

Comment:
I just realized a problem with this approach when thinking about PayPal Express: this function needs to work when called before and after checkout is complete. `receipt` is empty when this is called when the user checks out with PayPal, as this as they haven't paid yet, so this will throw a `TypeError`.